lib/logstash/filters/dns.rb

```diff
@@ -122,6 +122,12 @@ def resolve(event)
     @resolve.each do |field|
       is_array = false
       raw = event.get(field)
+
+      if raw.nil?
+        @logger.warn("DNS filter could not resolve missing field", :field => field)
```

Comment:
looking [at the original commit](https://github.com/elastic/logstash/pull/64/files#diff-d8dec0e7acf2c0940f8c4789ee0a30adR3), I cannot find a reason why this feature of resolving an array was introduced.

Looking at all of our other plugins like geoip and useragent, we only do a single field lookup, so I'd be tempted to drop the array support in this plugin too, but it's out of scope here.

> Since the failure in this case is due to null fields, I think its reasonable to expect that if the user does have that problem, that they could solve it with conditionals before or after this filter.

If the number of fields is variable (which is the purpose if this array handling) then it's really hard to remove nulls w/o a ruby filter.

Whenever I see a conversation reach this point my instinct is to provide the user the option to have either behaviour, maybe with a `abort_on_failure`/`continue_on_failure` or some other setting?